app/controllers/users_controller.rb

```diff
@@ -0,0 +1,18 @@
+class UsersController < ApplicationController
+  def index
+    # shows index view
+  end
+
+  def user
+    @user_id = params[:id]
+  end
+
+  def posts
+    @user_id = params[:id]
+  end
+
+  def post
+    @user_id = params[:id]
+    @post_id = params[:p_id]
+  end
```

Comment:
- [ ] These routes should not exist to begin with. Kindly remove these actions from the users controller. You should create a posts controller, just like this users controller and create `index` and `show` methods there.